### ISSUE TITLE
/opt/bitnami/scripts/libos.sh: line 335: 49 Segmentation fault

### ISSUE THREAD

**Issue by schroedt** [bitnami/charts#8349]
**Which chart**:
Bitnami MariaDB:10.4

**Describe the bug**
When restarting the pod, I get the following error:

```console
$ kubectl -n mariadb logs -f pod/release-10-4-mariadb-0
mariadb 08:51:02.89
mariadb 08:51:02.90 Welcome to the Bitnami mariadb container
mariadb 08:51:02.90 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 08:51:02.90 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 08:51:02.92
mariadb 08:51:02.92 INFO  ==> ** Starting MariaDB setup **
mariadb 08:51:02.99 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 08:51:03.00 INFO  ==> Initializing mariadb database
mariadb 08:51:03.02 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 08:51:03.02 INFO  ==> Using persisted data
mariadb 08:51:03.06 INFO  ==> Running mysql_upgrade
mariadb 08:51:03.07 INFO  ==> Starting mariadb in background
/opt/bitnami/scripts/libos.sh: line 335:    47 Segmentation fault      (core dumped) "$@" > /dev/null 2>&1
fabiansc@Kubernetes-Master:~$ kubectl -n mariadb delete pod/release-10-4-mariadb-0
pod "release-10-4-mariadb-0" deleted
fabiansc@Kubernetes-Master:~$ kubectl -n mariadb logs -f pod/release-10-4-mariadb-0
Error from server (BadRequest): container "mariadb" in pod "release-10-4-mariadb-0" is waiting to start: PodInitializing
fabiansc@Kubernetes-Master:~$ kubectl -n mariadb logs -f pod/release-10-4-mariadb-0
mariadb 08:53:08.11
mariadb 08:53:08.11 Welcome to the Bitnami mariadb container
mariadb 08:53:08.11 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 08:53:08.12 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 08:53:08.13
mariadb 08:53:08.13 INFO  ==> ** Starting MariaDB setup **
mariadb 08:53:08.19 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 08:53:08.21 INFO  ==> Initializing mariadb database
mariadb 08:53:08.25 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 08:53:08.25 INFO  ==> Using persisted data
mariadb 08:53:08.34 INFO  ==> Running mysql_upgrade
mariadb 08:53:08.38 INFO  ==> Starting mariadb in background
/opt/bitnami/scripts/libos.sh: line 335:    49 Segmentation fault      (core dumped) "$@" > /dev/null 2>&1

```

**To Reproduce**

1. deploy mariadb
2. wait a while
3. re-deply (upgrade)

**Expected behavior**
pod starts

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```console
$ helm repo add bitnami https://charts.bitnami.com/bitnami
$ helm repo update

$ helm uninstall release-10-4 --namespace mariadb

$ helm install release-10-4 bitnami/mariadb --namespace mariadb --set 'image.tag=10.4' --set 'auth.rootPassword=?????'
CHART VERSION: 9.7.0
APP VERSION: 10.5.12
```

**Additional context**

**Comment by javsalgar** [comment]
Hi,

Could you provide more details of your Kubernetes cluster? We test upgrades daily and this issue does not appear. Could it be due to memory or any PSP in place?

**Comment by schroedt** [comment]
Hi @javsalgar,

I am struggeling getting the PV size and usage. I guess the metrics are somehow not available.
I am running ceph, which auto-provisioned dynamic block storage. Ceph itself reports everything healthy.

Even though I could not find capacity usage information. Any recommendation for getting that info?

**Comment by schroedt** [comment]
Hi @javsalgar,

I setup a `prometheus` + `grafana` and was able to query the required data:
![grafik](https://user-images.githubusercontent.com/6430367/145714820-a0ab7f38-8f61-4819-bf55-cdf7a17efdd6.png)

Storage wise, it looks good, even though I got the segmentation fault:

![grafik](https://user-images.githubusercontent.com/6430367/145714865-441f3c60-bd77-4b9b-abf1-6fad86dcee76.png)

**Comment by alemorcuq** [comment]
It may be helpful to see when this segmentation fault is happening. Could you deploy it with the `BITNAMI_DEBUG` variable set to `true`? That should show extra info on the logs.

**Comment by schroedt** [comment]
I guess you mean `image.debug=true` from [bitnami/mariadb](https://github.com/bitnami/charts/tree/master/bitnami/mariadb)?

```
helm install release-10-4 bitnami/mariadb --namespace mariadb \
              --set image.tag=10.4 \
              --set auth.rootPassword=****** \
              --set image.debug=true
```

Log output is:

```
kubectl -n mariadb logs -f pod/release-10-4-mariadb-0
mariadb 15:44:33.30 
mariadb 15:44:33.30 Welcome to the Bitnami mariadb container
mariadb 15:44:33.31 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 15:44:33.31 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 15:44:33.31 
mariadb 15:44:33.32 INFO  ==> ** Starting MariaDB setup **
mariadb 15:44:33.36 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 15:44:33.37 INFO  ==> Initializing mariadb database
mariadb 15:44:33.38 DEBUG ==> Ensuring expected directories/files exist
mariadb 15:44:33.40 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 15:44:33.41 INFO  ==> Using persisted data
mariadb 15:44:33.46 INFO  ==> Running mysql_upgrade
mariadb 15:44:33.47 INFO  ==> Starting mariadb in background
2021-12-13 15:44:33 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.4.22-MariaDB) starting as process 49 ...
2021-12-13 15:44:33 0 [Note] InnoDB: Using Linux native AIO
2021-12-13 15:44:33 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2021-12-13 15:44:33 0 [Note] InnoDB: Uses event mutexes
2021-12-13 15:44:33 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-12-13 15:44:33 0 [Note] InnoDB: Number of pools: 1
2021-12-13 15:44:33 0 [Note] InnoDB: Using SSE2 crc32 instructions
2021-12-13 15:44:33 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2021-12-13 15:44:33 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
2021-12-13 15:44:33 0 [Note] InnoDB: Completed initialization of buffer pool
2021-12-13 15:44:33 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2021-12-13 15:44:33 0 [Note] InnoDB: The log sequence number 74608172 in the system tablespace does not match the log sequence number 74615308 in the ib_logfiles!
2021-12-13 15:44:33 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
2021-12-13 15:44:33 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-12-13 15:44:33 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-12-13 15:44:33 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-12-13 15:44:33 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-12-13 15:44:33 0 [Note] InnoDB: 10.4.22 started; log sequence number 74615308; transaction id 149285
2021-12-13 15:44:33 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2021-12-13 15:44:33 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-12-13 15:44:33 0 [Note] Server socket created on IP: '127.0.0.1'.
2021-12-13 15:44:33 0 [Warning] 'proxies_priv' entry '@% root@release-10-4-mariadb-0' ignored in --skip-name-resolve mode.
2021-12-13 15:44:33 0 [Note] Reading of all Master_info entries succeeded
2021-12-13 15:44:33 0 [Note] Added new Master_info '' to hash table
2021-12-13 15:44:33 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '10.4.22-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
2021-12-13 15:44:33 0 [Note] InnoDB: Buffer pool(s) load completed at 211213 15:44:33
Phase 1/7: Checking and upgrading mysql database
Processing databases
mysql
mysql.column_stats                                 OK
mysql.db                                           OK
mysql.event                                        OK
mysql.func                                         OK
mysql.global_priv                                  OK
mysql.gtid_slave_pos                               OK
mysql.help_category                                OK
mysql.help_keyword                                 OK
mysql.help_relation                                OK
mysql.help_topic                                   OK
mysql.index_stats                                  OK
mysql.innodb_index_stats                           OK
mysql.innodb_table_stats                           OK
mysql.plugin                                       OK
mysql.proc                                         OK
mysql.procs_priv                                   OK
mysql.proxies_priv                                 OK
mysql.roles_mapping                                OK
mysql.servers                                      OK
mysql.table_stats                                  OK
mysql.tables_priv                                  OK
mysql.time_zone                                    OK
mysql.time_zone_leap_second                        OK
mysql.time_zone_name                               OK
mysql.time_zone_transition                         OK
mysql.time_zone_transition_type                    OK
mysql.transaction_registry                         OK
Phase 2/7: Installing used storage engines... Skipped
Phase 3/7: Fixing views
211213 15:44:35 [ERROR] mysqld got signal 11 ;
This could be because you hit a bug. It is also possible that this binary
or one of the libraries it was linked against is corrupt, improperly built,
or misconfigured. This error can also be caused by malfunctioning hardware.

To report this bug, see https://mariadb.com/kb/en/reporting-bugs

We will try our best to scrape up some info that will hopefully help
diagnose the problem, but since we have already crashed, 
something is definitely wrong and this may fail.

Server version: 10.4.22-MariaDB
key_buffer_size=134217728
read_buffer_size=131072
max_used_connections=1
max_threads=153
thread_count=6
It is possible that mysqld could use up to 
key_buffer_size + (read_buffer_size + sort_buffer_size)*max_threads = 467769 K  bytes of memory
Hope that's ok; if not, decrease some variables in the equation.

Thread pointer: 0x7f9ba0000c08
Attempting backtrace. You can use the following information to find out
where mysqld died. If you see no messages after this, something went
terribly wrong...
stack_bottom = 0x7f9bf117be68 thread_stack 0x49000
/opt/bitnami/mariadb/sbin/mysqld(my_print_stacktrace+0x29)[0x56329e67ea89]
/opt/bitnami/mariadb/sbin/mysqld(handle_fatal_signal+0x53d)[0x56329e1e2e2d]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f9c08f31730]

Trying to get some variables.
Some pointers may be invalid and cause the dump to abort.
Query (0x7f9ba0012040): REPAIR NO_WRITE_TO_BINLOG VIEW `inventory_stock_1`

Connection ID (thread ID): 12
Status: NOT_KILLED

Optimizer switch: index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,index_merge_sort_intersection=off,engine_condition_pushdown=off,index_condition_pushdown=on,derived_merge=on,derived_with_keys=on,firstmatch=on,loosescan=on,materialization=on,in_to_exists=on,semijoin=on,partial_match_rowid_merge=on,partial_match_table_scan=on,subquery_cache=on,mrr=off,mrr_cost_based=off,mrr_sort_keys=off,outer_join_with_cache=on,semijoin_with_cache=on,join_cache_incremental=on,join_cache_hashed=on,join_cache_bka=on,optimize_join_buffer_size=on,table_elimination=on,extended_keys=on,exists_to_in=on,orderby_uses_equalities=on,condition_pushdown_for_derived=on,split_materialized=on,condition_pushdown_for_subquery=on,rowid_filter=on,condition_pushdown_from_having=on

The manual page at https://mariadb.com/kb/en/how-to-produce-a-full-stack-trace-for-mysqld/ contains
information that should help you find out what is causing the crash.
Writing a core file...
Working directory at /bitnami/mariadb/data
Resource Limits:
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            8388608              unlimited            bytes     
Max core file size        unlimited            unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             unlimited            unlimited            processes 
Max open files            1048576              1048576              files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       31461                31461                signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us        
Core pattern: |/usr/share/apport/apport %p %s %c %d %P %E

/opt/bitnami/scripts/libos.sh: line 335:    49 Segmentation fault      (core dumped) "$@"
/opt/bitnami/mariadb/bin/mysqlcheck: Got error: 2013: Lost connection to MySQL server during query when executing 'REPAIR NO_WRITE_TO_BINLOG VIEW ... '
FATAL ERROR: Upgrade failed
```

**Comment by spyro2000** [comment]
We have the very same problem with chart version 9.3.6. What did you do to fix this?

**Comment by schroedt** [comment]
Hi there,

We just started investigating. I do hope getting details from the maintainers soon. :-) 

Am 13. Dezember 2021 17:06:31 UTC schrieb spyro2000 ***@***.***>:
>We have the very same problem with chart version 9.3.6. What did you do to fix this?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/bitnami/charts/issues/8349#issuecomment-992685087
--

Mit freundlichen Grüßen

Fabian Schrödter

**Comment by spyro2000** [comment]
Thank you. Did did have to delete our persistent volume entirely. Seems, that MariaDB couldn't process some corrupted file (step 1/7 on it's repair process) and then just hard crashed.

**Comment by alemorcuq** [comment]
It looks like something is going wrong internally. My suggestion would be to post this on the MariaDB community, as we can't reproduce it and it doesn't look like an issue on our side.

**Comment by ChipWolf** [comment]
Also suffering from the same issue

**Comment by schroedt** [comment]
I guess I need to delete the rook-ceph PV and re-create it. Not sure what caused it.
As supervising storage on Kubernetes seem to be a story for itself I am still on it.. Kind of hard to get free/used PV spaces. Found something via Kubernetes Metcis + Prometheus, shifted to Grafana for visualization. However, I did not look deep enough into it.

 I guess I hit file limites breaking the whole image. Extension of a PV seems rather complicated.

Even though there isn't lots of things stored in the database (just one unused [magento](https://en.wikipedia.org/wiki/Magento) instance).
